compiler/rustc_session/src/lib.rs

```diff
@@ -24,3 +24,8 @@ pub use session::*;
 pub mod output;
 
 pub use getopts;
+
+/// Requirements for a `StableHashingContext` to be used in this crate.
+/// This is a hack to allow using the `HashStable_Generic` derive macro
+/// instead of implementing everything in `rustc_middle`.
```

Comment:
Is this really a hack? Even if it is one (and maybe especially if it is one) it would be good to add a more useful description here. Calling it a hack gives the impression that this is something brittle, not to be touched, and a piece of technical debt when actually it should be fine to do things this way, right?

Reply:
I copy-pasted this header from similar traits in other crates (e.g. `rustc_ast`, `rustc_target`). We could change those headers, but I think it would be better to do so all at once, in a follow-up PR.

Reply:
OK, fair enough.